### PR TITLE
refactor: HS-12345: Refactor Payment Element shimmer, Modal setState

### PR DESCRIPTION
**Pull request by jaggaroshu**
Updated set state
moved repeating Shimmer Ui to loop

### CODE REVIEW

File: src/Components/PaymentElementShimmer.res [hunk 1]
```diff
@@ -16,42 +16,30 @@ module Shimmer = {
...
-      </Shimmer>
-      <Shimmer>
-        <div className="animate-pulse w-auto h-12 rounded  bg-slate-200">
+module ElementShimmer = {
+  @react.component
+  let make = (~n=1) => {
+    Belt.Array.range(0, n - 1)
+    ->Js.Array2.map(index => {
+      <Shimmer key={index->Belt.Int.toString}>
+        <div className="animate-pulse w-full h-12 rounded bg-slate-200 ">
```
Comment (vsrivatsa-juspay) [inline src/Components/PaymentElementShimmer.res:25]:
```suggestion
        <div className="animate-pulse w-full h-12 rounded bg-slate-200">
```

File: src/Components/PaymentElementShimmer.res [hunk 2]
```diff
@@ -16,42 +16,30 @@ module Shimmer = {
...
           <div className="flex flex-row  my-auto">
```
Comment (vsrivatsa-juspay) [inline src/Components/PaymentElementShimmer.res:26]:
```suggestion
          <div className="flex flex-row my-auto">
```

File: src/Components/PaymentElementShimmer.res [hunk 3]
```diff
@@ -16,42 +16,30 @@ module Shimmer = {
...
             <div className=" w-10 h-5 rounded-full m-3 bg-white bg-opacity-70 " />
```
Comment (vsrivatsa-juspay) [inline src/Components/PaymentElementShimmer.res:27]:
```suggestion
            <div className="w-10 h-5 rounded-full m-3 bg-white bg-opacity-70" />
```

File: src/Components/PaymentElementShimmer.res [hunk 4]
```diff
@@ -16,42 +16,30 @@ module Shimmer = {
...
             <div className=" my-auto w-24 h-2 rounded m-3 bg-white bg-opacity-70 " />
```
Comment (vsrivatsa-juspay) [inline src/Components/PaymentElementShimmer.res:28]:
```suggestion
            <div className="my-auto w-24 h-2 rounded m-3 bg-white bg-opacity-70" />
```

File: src/Components/PaymentElementShimmer.res [hunk 5]
```diff
@@ -16,42 +16,30 @@ module Shimmer = {
...
       </Shimmer>
+    })
+    ->React.array
+  }
+}
+
+@react.component
+let make = () => {
+  <div className="flex flex-col gap-4">
+    <ElementShimmer n=2 />
```
Comment (vsrivatsa-juspay) [inline src/Components/PaymentElementShimmer.res:40]: if n is always going to be 2, it might as well not exist as a prop

**Review by prafulkoppalkar — APPROVED** [review]

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request closed without merging**